pkg/sdkserver/sdkserver_test.go

```diff
@@ -753,33 +719,51 @@ func TestSDKServerWatchGameServerFeatureSDKWatchSendOnExecute(t *testing.T) {
 		}
 	}
 
-	// if SDKWatchSendOnExecute feature is turned on, there are two stream.Send() calls should happen:
-	// one in sendGameServerUpdate, another one in WatchGameServer.
+	// There are two stream.Send() calls should happen: one in sendGameServerUpdate,
+	// another one in WatchGameServer.
 	assert.Equal(t, 2, totalSendCalls)
 }
 
 func TestSDKServerSendGameServerUpdate(t *testing.T) {
 	t.Parallel()
-
-	agruntime.FeatureTestMutex.Lock()
-	defer agruntime.FeatureTestMutex.Unlock()
-	err := agruntime.ParseFeatures(string(agruntime.FeatureSDKWatchSendOnExecute) + "=false")
-	require.NoError(t, err, "Can not parse FeatureSDKWatchSendOnExecute")
+	fixture := &agonesv1.GameServer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+		},
+		Status: agonesv1.GameServerStatus{
+			State: agonesv1.GameServerStateReady,
+		},
+	}
 
 	m := agtesting.NewMocks()
+	fakeWatch := watch.NewFake()
+	m.AgonesClient.AddWatchReactor("gameservers", k8stesting.DefaultWatchReactor(fakeWatch, nil))
 	sc, err := defaultSidecar(m)
 	require.NoError(t, err)
 	assert.Empty(t, sc.connectedStreams)
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	sc.ctx = ctx
+	sc.ctx = ctx
+	sc.informerFactory.Start(ctx.Done())
+
+	fakeWatch.Add(fixture.DeepCopy())
+	assert.True(t, cache.WaitForCacheSync(ctx.Done(), sc.gameServerSynced))
+	sc.gsWaitForSync.Done()
+
+	// wait for the GameServer to be populated, as we can't rely on WaitForCacheSync
+	require.Eventually(t, func() bool {
+		_, err := sc.gameServer()
+		return err == nil
+	}, time.Minute, time.Second, "Could not find the GameServer")
 
 	stream := newGameServerMockStream()
 	asyncWatchGameServer(t, sc, stream)
 	assert.Nil(t, waitConnectedStreamCount(sc, 1))
 
-	fixture := &agonesv1.GameServer{ObjectMeta: metav1.ObjectMeta{Name: "test-server"}}
+	// fixture := &agonesv1.GameServer{ObjectMeta: metav1.ObjectMeta{Name: "test-server"}}
```

Comment:
This is causing the linter to fail, since you shouldn't submit commented out code.